dali/test/python/operator_2/test_reduce.py

```diff
@@ -493,8 +493,8 @@ def _test_std_dev_large_data(rank, axes, device):
             ref = np.std(batch[i].astype(np.float64), axis=axes, ddof=0)
             assert np.allclose(out[i], ref, 1e-5, 1e-5)
 
-
-def test_std_dev_large_data():
+attr('slow')
+def slow_test_std_dev_large_data():
```

Comment:
Admittedly, this test is quite slow (18s on my machine), however, I'd keep one of the tests with "large data" - preferably this one. It's actually the only test that can quite reliably test batch subdivision mechanism and distribution of samples to different kernels. Note that it always uses one reduction (`sum`). Also, this test checks the numerical stability of summation and for that we need many numbers.

Reply:
BTW - I was able to cut the test time from 18s to 11s (for test with stddev) and from 12s to 8s (without stddev) by speeding up data generation:
```Python
def fast_large_random_batches(rank, batch_size, num_batches, lo=0, hi=1):
    max_extent = min(65536, int(np.floor(10000000**(1/rank))))

    # generate a maximum size buffer pre-filled with random numbers
    buf = np.random.uniform(low=lo, high=hi, size=max_extent**rank).astype(np.float32)

    data = []
    for _ in range(num_batches):
        batch = []
        for _ in range(batch_size):
            size = np.random.randint(1, max_extent, size=rank)
            vol = np.prod(size)
            # now that we know the actual volume of the sample, we can pick a random
            # location in the pre-filled buffer
            offset = np.random.randint(0, (buf.size - vol) + 1)
            # take a slice and reshape it to the desired shape - these are constant time operations
            sample = buf[offset:offset+vol].reshape(size)
            batch.append(sample)
        data.append(batch)
    return data
```

and the usage:
```Python
@nottest
def _test_reduce_large_data(rank, axes, device):
    batch_size = 16
    num_batches = 2
    data = fast_large_random_batches(rank, batch_size, num_batches)
```
and
```Python
@nottest
def _test_std_dev_large_data(rank, axes, device):
    batch_size = 16
    num_batches = 2
    data = fast_large_random_batches(rank, batch_size, num_batches, 1, 2);
```